.ci/static-checks.sh

```diff
@@ -600,7 +600,7 @@ static_check_docs()
 		# Git repo URL check
 		if echo "$url"|grep -q '^https.*git'
 		then
-			git ls-remote "$url" > /dev/null 2>&1 && continue
+			timeout "${KATA_NET_TIMEOUT}" git ls-remote "$url" > /dev/null 2>&1 && continue
```

Comment:
Nice. I wondered if there was a more git'ish way to do this - looks like we may be able to do a:
```
$ git config --global http.lowSpeedTime 30
```
or similar to set a timeout in git itself. I'm going to presume that will also propagate into `go get`. Worth a shot?

Reply:
Oh, this might go in hand with `http.lowSpeedLimit` as well..

Reply:
@grahamwhaley - the man page isn't clear to me on how those settings actually work - for example, what happens if you only set `http.lowSpeedTime`? What is the default for `http.lowSpeedLimit`? I like the idea, but unless you happen to know the specifics I think this might be a follow-on PR to optimise further.

Reply:
Yep, not very clear or obvious is it...:

https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables

> If the data rate of an HTTP operation is lower than GIT_HTTP_LOW_SPEED_LIMIT bytes per second for longer than GIT_HTTP_LOW_SPEED_TIME seconds, Git will abort that operation. These values override the http.lowSpeedLimit and http.lowSpeedTime configuration values.

Soo, I think we could set something that would effectively detect a dead, or near-dead, connection:

```bash
$ git config --global http.lowSpeedLimit 1000
$ git config --global http.lowSpeedTime 60
```

That is, if we have a data rate of less than 1k (it is not clear if that is 1k total, or 1k/s or ???, but, I think it does not matter really here....) for 1 minute, then git will abort.

I'm happy to fire a PR so we can see if it flies??